src/com/owncloud/android/services/observer/FileObserverService.java

```diff
@@ -276,16 +276,27 @@ private void addObservedFile(OCFile file, Account account) {
      */
     private void addObservedFile(String localPath, Account account) {
         File file = new File(localPath);
-        String parentPath = file.getParent();
-        FolderObserver observer = mFolderObserversMap.get(parentPath);
-        if (observer == null) {
-            observer = new FolderObserver(parentPath, account, getApplicationContext());
-            mFolderObserversMap.put(parentPath, observer);
-            Log_OC.d(TAG, "Observer added for parent folder " + parentPath + "/");
+
+        if(file.isDirectory()) {
+            File[] children = file.listFiles();
```

Comment:
And similarly, this will only add to observe the files that were downloaded before.

We need to extend ```FolderObserver.java``` so that can attend any file inside.